### DISCUSSION TITLE
Basic authentication using a password to the API

### DISCUSSION THREAD

**Discussion by YannBerry** [community/community#24335] [Apps, API and Webhooks]
<p>Hi,</p>
<p>Once a week I receive this email from github -&gt;</p>
<p><em>Hi <a class="mention" href="/yannberry">@YannBerry</a>,</em></p>
<p><em>On November 10th, 2020 at 10:37 (UTC) you used a password to access an endpoint through the GitHub API using curl:</em></p>
<p><em><a href="https://api.github.com/user/orgs" rel="noopener nofollow ugc">https://api.github.com/user/orgs</a></em></p>
<p><em>Basic authentication using a password to the API is deprecated and will soon no longer work. Visit <a href="https://developer.github.com/changes/2020-02-14-deprecating-password-auth/" rel="noopener nofollow ugc">https://developer.github.com/changes/2020-02-14-deprecating-password-auth/</a> for more information around suggested workarounds and removal dates.</em></p>
<p><em>Thanks,</em><br>
<em>The GitHub Team</em></p>
<p>As far as I’m concerned I don’t use the API. Why do I receive this email ? What should I do ?</p>
<p>Thanks !<br>
Yann</p>

**Comment by saya7852** [comment]
<p>I got that email just today.  Assume it is phishing?</p>

**Comment by YannBerry** [comment]
<p>I still don’t know. Please tell me if you find a solution :slight_smile:</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/yannberry">@YannBerry</a>, <a class="mention" href="/saya7852">@saya7852</a><br>
The link to 2020-02-14-deprecating-password-auth is a valid GitHub change notification.</p>
<p>The email looks valid (and similar to other users I have seen asking about, it is not asking you to click on any suspicious links.</p>
<p>You should switch to using personal access token, further details here</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token" target="_blank" rel="noopener nofollow ugc">Creating a personal access token - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>You mention “As far as I’m concerned I don’t use the API. Why do I receive this email ? What should I do ?”<br>
If that is the case I would change my password immediately (and enable 2FA on my account if not already enabled) as this warning email indicates your password credentials are being used by someone or some process (that you are unaware of) to list organizations for the authenticated user. Any such call will fail once you change your password. It also means that someone/some process could perform other actions as you as they have your password currently.</p>
<p>Yuu can change your password under account settings <a href="https://github.com/settings/security" rel="noopener nofollow ugc">https://github.com/settings/security</a><br>
You will also find ‘Sessions’ at this link, this is a list of devices that have logged into your account. Revoke any sessions that you do not recognize.</p>
<p>You could also review <a href="https://github.com/settings/applications" rel="noopener nofollow ugc">https://github.com/settings/applications</a> for installed GitHub Apps / Authorized GitHub Apps / Authorized OAuth Apps, and clean up anything not needed/trusted.</p>
<p>You can also review/monitor your accounts Security log for auditable actions you have not performed</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/settings/security-log" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:500/500;"><img src="https://user-images.githubusercontent.com/39440862/181084271-8120c5d5-2bc3-444c-a324-8b9d16172490.png" class="thumbnail" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png"></div>

<h3><a href="https://github.com/settings/security-log" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

<p><a href="https://github.com/settings/security-log" target="_blank" rel="noopener nofollow ugc">ngs/security-log</a></p>

  <p><span class="label1">GitHub is where people build software. More than 50 million people use GitHub to discover, fork, and contribute to over 100 million projects.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<p>
such as<br>
USERNAME – user.failed_login<br>
Failed to login</p>

**Comment by saya7852** [comment]
<p>Thanks for your detailed explanation <a class="mention" href="/byrneh">@byrneh</a>!  Suppose nne should not ALWAYS assume anything with a hyperlink as phishing :sweat_smile:</p>

**Comment by YannBerry** [comment]
<aside class="quote no-group" data-username="byrneh" data-post="4" data-topic="142959">
<div class="title">
<div class="quote-controls"></div>
 byrneh:</div>
<blockquote>
<p>acces</p>
</blockquote>
</aside>
<p><a class="mention" href="/byrneh">@byrneh</a> Thank you for your detailed answer :slight_smile:</p>

**Comment by McMike4** [comment]
Hi,
I am trying to use this below:

curl -X GET --user username:password -i http://yourdomain.com/wp-json/wp/v2/posts?status=draft

But I’ve got an 400 error message (HTTP/1.1 400 Bad request). What could that be? I’ve already tried other so far but this one doesn’t work…

Thanks in advance
